connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/builder/parts/ApacheRequestUriBuilder.java

```diff
@@ -21,14 +21,26 @@
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URL;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
 import org.apache.hc.core5.http.io.support.ClassicRequestBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ApacheRequestUriBuilder implements ApacheRequestPartBuilder {
+  private static final Logger LOG = LoggerFactory.getLogger(ApacheRequestUriBuilder.class);
 
   @Override
   public void build(ClassicRequestBuilder builder, HttpCommonRequest request) {
     try {
-      var url = new URL(request.getUrl());
+      // We try to decode the URL first, because it might be encoded already
+      // which would lead to double encoding. Decoding is safe here, because it does nothing if
+      // the URL is not encoded.
+      var decodedUrl =
+          URLDecoder.decode(
+              Optional.ofNullable(request.getUrl()).orElse(""), StandardCharsets.UTF_8);
+      var url = new URL(decodedUrl);
       builder.setUri(
           // Only this URI constructor escapes the URL properly
           new URI(
```

Comment:
For easier maintenance and testability, could you extract this into a static utility method?